appservice/consumers/roomserver.go

```diff
@@ -159,5 +159,33 @@ func (s *OutputRoomEventConsumer) appserviceIsInterestedInEvent(ctx context.Cont
 		}).WithError(err).Errorf("Unable to get aliases for room")
 	}
 
+	// Check if any of the members in the room match the appservice
+	// TODO: appserviceIsInterestedInEvent should be wrapped in a cache
+	// as this part can involve a lot of DB lookups
+	membershipReq := api.QueryMembershipsForRoomRequest{
+		JoinedOnly: true,
+		RoomID:     event.RoomID(),
+	}
+	var membershipRes api.QueryMembershipsForRoomResponse
+
+	// XXX: This should be membership at the time of the event, not current membership. I'm not sure
+	// how to extract this though, help?!
+	// TODO: We also don't need to get all the members. Could we query the DB directly with a set of
```

Comment:
I leave this up to the maintainers on the urgency of this, but obviously each event going through dendrite will currently check the membership of a room. If there is an easy way to cache *this* function call and invalidate it when the membership (or aliases) changes, then let's do that.